### PR TITLE
Rename dispatcher to executor

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR renames dispatcher to executor, and makes the dispatcher name available.

### CODE REVIEW

File: cpp/src/Ice/ConnectionI.cpp [hunk 1]
```diff
@@ -2089,7 +2089,7 @@ Ice::ConnectionI::ConnectionI(
       _connector(connector),
       _endpoint(endpoint),
       _adapter(adapter),
-      _dispatcher(_instance->initializationData().dispatcher), // Cached for better performance.
+      _hasExecutor(_instance->initializationData().executor), // Cached for better performance.
```
Comment (pepone) [inline cpp/src/Ice/ConnectionI.cpp:2092]: Keep the comments aligned

File: cpp/test/Glacier2/sessionHelper/Client.cpp [hunk 2]
```diff
@@ -235,11 +235,11 @@ Client::run(int argc, char** argv)
     _initData.properties = Ice::createProperties(argc, argv, communicator->getProperties());
     _initData.properties->setProperty("Ice.Default.Router", "Glacier2/router:" + getTestEndpoint(50));
 
-    Dispatcher dispatcher;
-    auto dfut = std::async(launch::async, [&dispatcher] { dispatcher.run(); });
+    Executor executor;
+    auto dfut = std::async(launch::async, [&executor] { executor.run(); });
```
Comment (pepone) [inline cpp/test/Glacier2/sessionHelper/Client.cpp:239]: We should rename `dfut` variable to something more meaningful. `executorFuture`?

**Review by pepone — APPROVED** [review]

File: cpp/src/Ice/ConnectionI.cpp (hunk 1 above)
Comment (bernardnormier) [reply cpp/src/Ice/ConnectionI.cpp:2092]: Fixed

File: cpp/test/Glacier2/sessionHelper/Client.cpp (hunk 2 above)
Comment (bernardnormier) [reply cpp/test/Glacier2/sessionHelper/Client.cpp:239]: Fixed

File: cpp/src/Ice/ThreadPool.h [hunk 3]
```diff
@@ -126,8 +127,8 @@ namespace IceInternal
         const InstancePtr _instance;
 #ifdef ICE_SWIFT
         const dispatch_queue_t _dispatchQueue;
-#else // Ice for Swift does not support a dispatcher
-        std::function<void(std::function<void()>, const std::shared_ptr<Ice::Connection>&)> _dispatcher;
+#else // Ice for Swift does not support a executor
```
Comment (externl) [inline cpp/src/Ice/ThreadPool.h:130]:
```suggestion
#else // Ice for Swift does not support an executor
```

**Review by externl — APPROVED** [review]

**Pull request merged**